### ISSUE TITLE
Projectile doesn't complete when pressing tab after `M-x projectile-find-file`

### ISSUE THREAD

**Issue by slipset** [bbatsov/projectile#1724]
## Expected behavior
When in a project, executing `M-x projectile-find-file RET <some-file-prefix> TAB I expect to be able to choose the files that match.

I remember having some trouble with this previously due to not having helm/ido or some such installed.

## Actual behavior
Projectile responds with `[No match]` even for files I know exist.

Doing `M-x projectile-find-file RET TAB` shows a buffer with all completions

## Steps to reproduce the problem

As described above

### Projectile version information

`M-x projectile-version` gives `Projectile nil`
My projectile.el is stored under `elpa/projectile-20211116.700/` which might give a better indication of the version.

### Emacs version

GNU Emacs 27.2 (build 1, aarch64-apple-darwin21.1.0, Carbon Version 165 AppKit 2113)
 of 2021-11-20

### Operating system

macOS Monterey 12.0.1

**Comment by bbatsov** [comment]
Which completion method are you using https://docs.projectile.mx/projectile/configuration.html#completion-options ?

**Comment by slipset** [comment]
So, this is solved by setting
```
(setq projectile-completion-system 'ido)
```

The value it had before doing this was `auto`

**Comment by slipset** [comment]
I probably use a completion system, but I never configured one, so I'm probably using whatever default eMacs provides.

**Comment by bbatsov** [comment]
I guess you hadn't enabled `ido-mode` globally, otherwise auto would have picked this up. The default Emacs completion, which got selected instead, is less sophisticated, as you've noticed yourself. :-)

**Comment by slipset** [comment]
We might have discussed this before, but when I started using projectile some time ago, I believe the default completion system must have been `Ido`. I don't think I would ever have started using projectile if the default completion system was, well the default one.

So, I would gently suggest to return to the old default where a naive user like myself would get ido, rather than the, as you put it yourself, the less sophisticated, default system.

Of course, one could argue that I should just RTFM and get on with my life :)

**Comment by bbatsov** [comment]
Yeah, it's true that Projectile used `ido` by default originally. Back then it was clearly the best option possible, but today it's using "auto" to respect whatever completion system the user favors, as there are a ton of options out there. It's a bit weird to default to `ido` if someone has enabled `ivy-mode` or `selectrum-mode` for instance. 

In your case, my expectation was simply that if someone likes `ido`, they probably enabled `ido-mode` globally and Projectile would pick this up and use `ido` automatically. Perhaps it's just availability bias, but I thought that pretty much everyone replaces the default completion in their personal config.

**Comment by aij** [comment]
I ran into this too when I updated `projectile` for compatibility with Emacs 28. I didn't realize it was because of the completion system until I did a git bisect and tracked it down to 5410ae0cedeebc76b1cfd10231243182e6f11993 "Add auto-detection of completion system (#1602)"

FWIW, I never configured a default completion system, in part because I found different packages tend to use the one that works best for them. That was over 5 years ago though, so maybe now the various completion systems are more standardized/interchangeable... I guess I should re-evaluate.
